File: 62-unique-paths/unique-paths.go
func uniquePaths(m int, n int) int {
    //ones for first row/col
    grid := make([][]int, m)
    for i := range(grid) {
        grid[i] = make([]int, n)
    }
    for i := 0; i < n; i++ {
        grid[0][i] = 1
    }

    for i := 0; i < m; i++ {
        grid[i][0] = 1
    }

    //DP from top left down to bottom right
    for i := 1; i < len(grid); i++ {
        for j := 1; j < len(grid[0]); j++ {
            grid[i][j] = grid[i-1][j] + grid[i][j-1]
        }
    }
    return grid[m-1][n-1]
}

File: 62-unique-paths/unique-paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] dp = new int[m][n];
        //First row
        for(int col = 0; col  < dp[0].length; col++) dp[0][col] = 1;
        //First col
        for(int[] row : dp) row[0] = 1;

        for(int row = 1; row < dp.length; row++) {
            for(int col = 1; col < dp[0].length; col++) {
                dp[row][col] = dp[row - 1][col] + dp[row][col - 1];
            }
        }
        return dp[m - 1][n - 1];
    }
}